subsys/bluetooth/services/fast_pair/fp_storage/include/fp_storage_ak_bond.h

```diff
@@ -0,0 +1,143 @@
+ * @defgroup fp_storage_ak_bond Fast Pair storage of associations between Account Keys and BLE bonds
+ * @brief Internal API for Fast Pair storage of associations between Account Keys and BLE bonds
+ *
+ * Used only when the CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND Kconfig option is enabled.
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Create and save the Account Key association with the potential Bluetooth bond that is being
+ *  established from the active connection.
+ *
+ *  The module links the Account Key information with the Bluetooth bond and stores
+ *  it in the NVM memory. The association and the Bluetooth bond will be deleted if the bond is not
+ *  confirmed using the @ref fp_storage_ak_bond_conn_confirm API before the bonding process is
+ *  finalized using the @ref fp_storage_ak_bond_conn_finalize API and also, in case of the Fast Pair
+ *  initial pairing procedure, if the Account Key is not written by the conn before the
+ *  the @ref fp_storage_ak_bond_conn_finalize API is called.
+ *
+ * @param[in] conn_ctx Connection context associated with the potential the Bluetooth bond.
+ * @param[in] addr The Bluetooth address of the potential bond.
+ * @param[in] account_key Account Key that the bond is associated with. It can be set to NULL in
+ *			  case of the initial pairing Procedure, when the Account Key is not known
+ *			  yet.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_storage_ak_bond_conn_create(const void *conn_ctx, const bt_addr_le_t *addr,
+				   const struct fp_account_key *account_key);
+
+/** Indicate that the potential Bluetooth bond has been confirmed.
+ *
+ *  It confirms the potential Bluetooth bond associated with the Account Key. The association should
+ *  be earlier created using the @ref fp_storage_ak_bond_conn_create API. In case of the Fast Pair
+ *  initial pairing procedure, the Account Key write is required to happen after calling this
+ *  function to fully confirm the association between Account Key and BLE bond.
+ *
+ * @param[in] conn_ctx Connection context associated with the Bluetooth bond.
+ * @param[in] addr The Bluetooth address of the bond.
+ */
+void fp_storage_ak_bond_conn_confirm(const void *conn_ctx, const bt_addr_le_t *addr);
+
+/** Indicate that the bonding process has been finalized.
+ *
+ *  It relates to the Account Key association with the Bluetooth bond that was created
+ *  earlier using the @ref fp_storage_ak_bond_conn_create API. This function should be called when
+ *  the BLE conn disconnects to let the storage subsystem handle incomplete bonds.
+ *
+ *  If the @ref fp_storage_ak_bond_conn_create API has not been called for this conn earlier,
+ *  the call is ignored.
+ *
+ * @param[in] conn_ctx Connection context associated with the Bluetooth bond.
+ */
+void fp_storage_ak_bond_conn_finalize(const void *conn_ctx);
+
+/** Update the address of the Bluetooth bond associated with the Account Key.
```

Comment:
`update` -> `resolve` ? To make it more straightforward

Reply:
This function is not really resolving anything. It updates the address with the address that was already resolved earlier.

Reply:
Ok, let's keep the naming from perspective of the `fp_storage`. Thanks for adding the comment